### DISCUSSION TITLE
desktop-starter ModelSelectorWidgetControl

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#2382] [Q&A] [answered]
i want show the model and categories selec tree
in the MainFrontstage.tsx for desktop-starter project code :

 bottomLeft={
          <Zone defaultState={ZoneState.Open} allowsMerging={true}
            widgets={[
              <Widget id="ModelSelectorWidgetControl" control={ModelSelectorWidgetControl} defaultState={WidgetState.Closed} fillZone={true}
                iconSpec="icon-properties-list" labelKey="App:components.ModelSelectorWidgetControl"
                applicationData={{
                  iModelConnection: UiFramework.getIModelConnection(),
                }}
                syncEventIds={[SyncUiEventId.SelectionSetChanged]}
                stateFunc={this._determineWidgetStateForSelectionSet}
              />,
            ]}
          />
        }

![22222222222222222](https://user-images.githubusercontent.com/18022720/135424025-5ebd63dc-f9d2-4a57-882c-b43cb53d56d7.png)

**Comment by williamkbentley** [comment]
~~Hello,~~

~~I copied your code into my desktop-starter instance, but I was not able to reproduce that presentation error:~~

![image](https://user-images.githubusercontent.com/38542573/135494848-f0853d70-7d65-4776-adc0-246446037373.png)

~~What version of iModel.js are you using? Mine is v2.17.1.~~

I **am** able to reproduce this issue on first launch of `npm run start`. I'm investigating the initialization steps to make sure that IPC is registered correctly.

**Comment by williamkbentley** [comment, marked as answer]
In main.ts, add `useSingleManager: true,` to your Presentation.initialize. 

@grigasp @saskliutas What does `useSingleManager` mean for an Electron app? Surely there will only be one manager for a desktop app.

**Reply by grigasp** [reply]
Presentation backend may work in two modes: single manager and multiple managers. The latter is intended for cloud cases, when the backend has to handle multiple frontends - then each frontend gets its own `PresentationManager` on the backend. In case of electron, since there's only one frontend anyway, both modes result in only one `PresentationManager` instance. However, the difference is that with `useSingleManager = true` we create the manager when `Presentation.initialize` is called and otherwise it's created only on first presentation request.

The problem here is that we're registering our IPC handled in a wrong place - that should be happening during `Presentation.initialize` instead of when a `PresentationManager` instance is created. I have a fix for master here: #2398.

Setting `useSingleManager` to true when calling `Presentation.initialize` is a good workaround for this, so I don't think we need to backport the fix to 2.19.

**Reply by williamkbentley** [reply]
Great, thanks
